winton_kafka_streams/processor/_stream_thread.py

```diff
@@ -146,7 +146,7 @@ def run(self):
 
             self.log.debug('Ending stream thread...')
         finally:
-            self.commit_all()
+            self.commit_all()  # TODO does this mean we update the consumer offset under a failure scenario?
```

Comment:
I think this should be fine, as this will only commit offsets that have been stored by successful processing. So a tuple is processed, offset for that is updated in-process, and then occasionally the in-process offsets are committed back to Kafka.

Reply:
Thanks